q2_dwq2/_types_and_formats.py

```diff
@@ -0,0 +1,70 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2024, Greg Caporaso.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from skbio import DNA
+from skbio.io import UnrecognizedFormatError
+
+from qiime2.plugin import SemanticType, TextFileFormat, model, ValidationError
+
+# Define semantic types
+SingleDNASequence = SemanticType("SingleDNASequence")
+
+
+# Define formats
+class SingleRecordDNAFASTAFormat(TextFileFormat):
```

Comment:
Something we could consider, but I don't know if this is a good idea or not, would be to use semantic validation on the regular fasta format. Downside is it over-commits the transformer to work with skbio.DNA which is perhaps unexpected.

So maybe this is the right approach, but we could use validation for the fact that it is DNA. Although that may be a nice refactoring down the road if you extend this for proteins etc

Reply:
I was thinking of refactoring to work with proteins in the future, possibly to incorporate use of `TypeMap`. So let's come back to that.